iothub/device/src/IotHubBaseClient.cs

```diff
@@ -489,7 +511,15 @@ internal void OnDesiredStatePatchReceived(DesiredProperties patch)
             if (Logging.IsEnabled)
                 Logging.Info(this, patch.GetSerializedString(), nameof(OnDesiredStatePatchReceived));
 
-            _ = _desiredPropertyUpdateCallback.Invoke(patch);
+            try
+            {
+                _ = _desiredPropertyUpdateCallback.Invoke(patch);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
```

Comment:
We're not awaiting on this task, so this exception block wouldn't actually catch anything. The code above is simply queuing up a task that runs on an unmonitored thread.